src/net_processing.cpp

```diff
@@ -5937,12 +5937,12 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
             // If a snapshot chainstate is in use, we want to find its next blocks
             // before the background chainstate to prioritize getting to network tip.
             FindNextBlocksToDownload(*peer, get_inflight_budget(), vToDownload, staller);
-            if (m_chainman.BackgroundSyncInProgress() && !IsLimitedPeer(*peer)) {
+            auto historical_blocks = m_chainman.GetHistoricalBlockRange();
+            if (historical_blocks && !IsLimitedPeer(*peer)) {
                 TryDownloadingHistoricalBlocks(
                     *peer,
                     get_inflight_budget(),
-                    vToDownload, m_chainman.GetBackgroundSyncTip(),
-                    Assert(m_chainman.GetSnapshotBaseBlock()));
+                    vToDownload, historical_blocks->first, historical_blocks->second);
```

Comment:
If practical, it would be nice to have this change in a seperate commit from the (simpler) changes above.